src/main/java/careconnect/model/person/UniquePersonList.java

```diff
@@ -39,13 +39,15 @@ public boolean contains(Person toCheck) {
     /**
      * Adds a person to the list.
      * The person must not already exist in the list.
+     * Sorts the list by name lexicographically.
      */
     public void add(Person toAdd) {
         requireNonNull(toAdd);
         if (contains(toAdd)) {
             throw new DuplicatePersonException();
         }
-        internalList.add(toAdd);
+        this.internalList.add(toAdd);
+        this.internalList.sort((p1, p2) -> p1.getName().toString().compareTo(p2.getName().toString()));
```

Comment:
```suggestion
        this.sortListLexographically();
```

Reply:
A possible optimization is to do a binary search then add at a specific index using the overloaded method `add(int index, E element)`.